doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -704,57 +688,33 @@ the conversion in hypotheses :n:`{+ @ident}`.
             Goal 1 + 0 = 1.
             unfold Nat.add.
 
-   .. tacv:: unfold @qualid in @goal_occurrences
-
-      Replaces :n:`@qualid` in hypothesis (or hypotheses) designated
-      by :token:`goal_occurrences` with its definition and replaces
-      the hypothesis with its :math:`\beta`:math:`\iota` normal form.
-
-   .. tacv:: unfold {+, @qualid}
-
-      Replaces :n:`{+, @qualid}` with their definitions and replaces
-      the current goal with its :math:`\beta`:math:`\iota` normal
-      form.
-
-   .. tacv:: unfold {+, @qualid at @occurrences }
-
-      The list :token:`occurrences` specify the occurrences of
-      :n:`@qualid` to be unfolded. Occurrences are located from left
-      to right.
-
       .. exn:: Bad occurrence number of @qualid.
          :undocumented:
 
       .. exn:: @qualid does not occur.
          :undocumented:
 
-   .. tacv:: unfold @string
-
-      If :n:`@string` denotes the discriminating symbol of a notation
-      (e.g. "+") or an expression defining a notation (e.g. `"_ +
-      _"`), and this notation denotes an application whose head symbol
-      is an unfoldable constant, then the tactic unfolds it.
-
-   .. tacv:: unfold @string%@ident
-
-      This is variant of :n:`unfold @string` where :n:`@string` gets
-      its interpretation from the scope bound to the delimiting key
-      :token:`ident` instead of its default interpretation (see
-      :ref:`Localinterpretationrulesfornotations`).
+.. tacn:: fold {+ @one_term } {? @occurrences }
 
-   .. tacv:: unfold {+, {| @qualid | @string{? %@ident } } {? at @occurrences } } {? in @goal_occurrences }
+   .. todo PR: following is wrong because you can have multiple one_terms.  How
+      do multiple one_terms interact with occurrences?? I suppose the one_term is
+      a pattern match (& should be a one_pattern?).  What should this say?
 
-      This is the most general form.
+      Also "This tactic is particularly useful when a fixpoint definition has been
+      wrongfully unfolded" seems specific to coqtop.  In CoqIDE or another GUI, you'd
+      just undo the last tactic.  So OK to drop that?
 
-.. tacn:: fold @term
-   :name: fold
-
-   This tactic applies to any goal. The term :n:`@term` is reduced using the
-   :tacn:`red` tactic. Every occurrence of the resulting :n:`@term` in the goal is
-   then replaced by :n:`@term`. This tactic is particularly useful when a fixpoint
+   Applies delta-expansion (the inverse of :term:`delta-reduction`) to the specified
+   :n:`@one_term`\s.  If :n:`@occurrences` is given,
```

Comment:
Please revert this change.  The original text was accurate and precise, while the new text is wrong (delta-expansion is not applied, `fold` is much stronger than just inverting delta, and the folding happens in the goal/hypotheses, not in the argument to `fold`, which is supposed to be the final folded result, not the unfolded version).

Reply:
Got it with some minor improvements.  This one works a bit differently than the other conversions in that you specify the resulting subterm in the tactic rather than the subterm you want to match.